build-aux/snap/snapcraft.yaml

```diff
@@ -87,6 +87,8 @@ parts:
           git diff
         ) > $SNAPCRAFT_PART_INSTALL/usr/lib/snapd/dirty-git-tree-info.txt
       fi
+      mkdir -p $SNAPCRAFT_PART_INSTALL/scripts
+      cp -a $SNAPCRAFT_PROJECT_DIR/scripts/* $SNAPCRAFT_PART_INSTALL/scripts
```

Comment:
I tried it, but it didn't work. You must call the original script from outside the snap container.

Reply:
Applications like Firefox are using `core22` as a base, so would still be calling the proxies that talk to `snap userd` or `xdg-desktop-portal`.

What I'm suggesting is that we stop shipping the proxies in the `core22-desktop` snap specifically, so `/usr/bin/xdg-open` and `/usr/bin/xdg-settings` will be the original scripts from the point of view of `snap userd`. The snaps that actually use `core22-desktop` as a base are not relying on those scripts (or in the case of `xdg-open`, should still work because of the desktop-launch plug).

Reply:
I'll test it...

Reply:
Ok, it is Core22 the one that has the modified xdg-open and xdg-settings scripts...

Reply:
I mean: in Core22, xdg-open is:

```
#!/bin/sh
exec snapctl user-open "$@"
```

and xdg-settings contains several options, but basically calls a DBus method in `io.snapcraft.Settings` with the desired parameters. Thus both scripts in Core22 expect the user snap daemon to call the original scripts.

Reply:
You was right. With this other patch, all works fine https://github.com/canonical/core-base-desktop/pull/24